dumux-precice/couplingadapter.cc

```diff
@@ -310,111 +179,63 @@ bool CouplingAdapter::isCoupledEntity(const int faceID) const
+    return combinedKey;
 }
 
 void CouplingAdapter::print(std::ostream &os)
 {
     os << indexMapper_;
 }
 
-bool CouplingAdapter::checkIfActionIsRequired(const std::string &condition)
-{
-    assert(wasCreated_);
-    return precice_->isActionRequired(condition);
-}
-
-void CouplingAdapter::actionIsFulfilled(const std::string &condition)
-{
-    assert(wasCreated_);
-    precice_->markActionFulfilled(condition);
-}
-
-void CouplingAdapter::readBlockDataFromPrecice(const int dataID,
-                                               std::vector<double> &data,
-                                               const QuantityType quantity_type)
+void CouplingAdapter::readQuantityFromOtherSolver(
+    const precice::string_view &meshName,
+    const precice::string_view &dataName,
+    double relativeReadTime)
 {
-    assert(wasCreated_);
-    if (quantity_type == QuantityType::Scalar) {
-        assert(vertexIDs_.size() == data.size());
-        precice_->readBlockScalarData(dataID, vertexIDs_.size(),
-                                      vertexIDs_.data(), data.data());
-    } else {
-        assert(vertexIDs_.size() * getDimensions() == data.size());
-        precice_->readBlockVectorData(dataID, vertexIDs_.size(),
-                                      vertexIDs_.data(), data.data());
-    }
+    precice::span<double> dataValuesSpan(getQuantityVector(meshName, dataName));
+    precice_->readData(meshName, dataName, vertexIDsSpan_, relativeReadTime,
+                       dataValuesSpan);
 }
 
-void CouplingAdapter::writeBlockDataToPrecice(const int dataID,
-                                              std::vector<double> &data,
-                                              const QuantityType quantity_type)
+void CouplingAdapter::writeQuantityToOtherSolver(
+    const precice::string_view &meshName,
+    const precice::string_view &dataName)
 {
-    assert(wasCreated_);
-    if (quantity_type == QuantityType::Scalar) {
-        assert(vertexIDs_.size() == data.size());
-        precice_->writeBlockScalarData(dataID, vertexIDs_.size(),
-                                       vertexIDs_.data(), data.data());
-    } else {
-        assert(vertexIDs_.size() * getDimensions() == data.size());
-        precice_->writeBlockVectorData(dataID, vertexIDs_.size(),
-                                       vertexIDs_.data(), data.data());
-    }
+    precice::span<const double> dataValuesSpan(
+        getQuantityVector(meshName, dataName));
+    precice_->writeData(meshName, dataName, vertexIDsSpan_, dataValuesSpan);
 }
 
 bool CouplingAdapter::hasToWriteInitialData()
 {
     assert(wasCreated_);
-    return checkIfActionIsRequired(
-        precice::constants::actionWriteInitialData());
-}
-
-void CouplingAdapter::announceInitialDataWritten()
-{
-    assert(wasCreated_);
-    precice_->markActionFulfilled(precice::constants::actionWriteInitialData());
+    return precice_->requiresInitialData();
 }
 
 bool CouplingAdapter::hasToReadIterationCheckpoint()
```

Comment:
```suggestion
bool CouplingAdapter::hasToReadCheckpoint()
```

Reply:
I would suggest renaming to `requiresToReadCheckPoint` to keep in consistence with the `requiresToWriteInitialData` above. Same for the next rename. Would this be better?  :)

Reply:
Yes. The closer we are to the original API names, the more consistent it is.